infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -1,16 +1,12 @@
 package org.corfudb.infrastructure;
 
-import com.github.benmanes.caffeine.cache.Cache;
-import com.github.benmanes.caffeine.cache.CacheWriter;
-import com.github.benmanes.caffeine.cache.Caffeine;
-import com.github.benmanes.caffeine.cache.RemovalCause;
-import com.google.common.annotations.VisibleForTesting;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import org.corfudb.runtime.view.Address;
-
-import javax.annotation.Nonnull;
+import java.util.Comparator;
```

Comment:
This is a new data structure, please add specific tests that test eviction logic. So, a test to verify that the window can keep sliding, and eviction of a version will remove multiple keys, etc.

Reply:
Added more tests.